test/shell/test_scalafmt.sh

```diff
@@ -0,0 +1,58 @@
+# shellcheck source=./test_runner.sh
+dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+. "${dir}"/test_runner.sh
+. "${dir}"/test_helper.sh
+runner=$(get_test_runner "${1:-local}")
+
+run_non_default_formatting() {
```

Comment:
why is it non_default_formatting? I don't see anything customizing the scalafmt conf file.
Speaking of this maybe we need a test to demonstrate passing in a custom scalafmt file?

Reply:
Since it's an extension, we don't format code by default.

Agree on having a test to demonstrate passing a custom scalafmt conf file.

Reply:
Test added.

Reply:
WDYT about renaming to `run_formatting()`? What's the value of saying it's non-default? Are you concerned someone will think this happens by default? Why? Not saying you're wrong but I don't see the issue

Reply:
Also like on the custom conf test

Reply:
I agree on your comment. Renamed.